File: Week 1/10.Multiple_input.java
/*
 * WEEK 1 - JAVA FUNDAMENTALS
 * Topic: Reading Multiple Input Types
 * File: 10.Multiple_input.java
 *
 * CONCEPT:
 * Demonstrates reading multiple data types (int and String) in sequence
 * from user input. Also showcases the Scanner class methods reference
 * for all supported data types.
 *
 * KEY POINTS:
 * - Different Scanner methods can be chained to read mixed-type input
 * - next() reads a single token (word); nextLine() reads the full line
 * - nextByte(), nextShort(), nextInt(), nextLong() for integer types
 * - nextFloat(), nextDouble() for floating point types
 * - print() and println() can be mixed for formatting output
 * - print(a) followed by println(str) prints them on the same line with
 *   a newline only after str
 *
 * SYNTAX:
 * int a = s.nextInt();    // read integer token
 * String str = s.next();  // read string token
 */

package fundamentals;
import java.util.Scanner;
public class ArithematicOperators {
    public static void main(String args[]) {


        Scanner s = new Scanner(System.in);
        int a = s.nextInt();    // Read an integer from input
        String str = s.next();  // Read the next word/token as a String
        System.out.print(a);    // Print integer (no newline)
        System.out.println(str);  // Print string (with newline) - appears on same line as 'a'
    }
}

/*


Other scanner options


Some commonly used Scanner class methods are as follows:
METHOD                                            DESCRIPTION
public String next()                    It returns the next token from the Scanner.
public String nextLine()                It moves the Scanner position to the next line and returns the value as a string.
public byte nextByte()                  It scans the next token as a byte.
public short nextShort()                It scans the next token as a short value.
public int nextInt()                    It scans the next token as an int value.
public long nextLong()                  It scans the next token as a long value.
public float nextFloat()                It scans the next token as a float value.
public double nextDouble()              It scans the next token as a double value.



*/

File: Week 1/rust/10.multiple_input.rs
/*
 * WEEK 1 - RUST FUNDAMENTALS
 * Topic: Reading Mixed-Type Input
 * File: 10.multiple_input.rs
 *
 * CONCEPT:
 * Read an integer and a string from stdin and print them on the same line.
 * Demonstrates parsing different types from the same line of input.
 *
 * KEY POINTS:
 *  - `split_whitespace()` yields token slices.
 *  - Each `parse::<T>()` call needs the destination type (or explicit turbofish).
 *  - `print!(...)` keeps the cursor on the same line; finish with `println!()`.
 *
 * SYNTAX:
 *   let mut line = String::new();
 *   io::stdin().read_line(&mut line)?;
 *   let mut it = line.split_whitespace();
 *   let i: i32 = it.next().unwrap().parse().unwrap();
 *   let s: &str = it.next().unwrap();
 *
 * DRY RUN:
 *  Stdin: "42 hello"
 *    i=42, s="hello"
 *    -> "42hello"  then  "42 hello"
 */

use std::io;

fn main() -> io::Result<()> {
    let mut line = String::new();
    io::stdin().read_line(&mut line)?;
    let mut it = line.split_whitespace();

    let i: i32 = it.next().expect("missing int").parse().expect("not an int");
    let s: String = it.next().unwrap_or("").to_string();

    // No newline between -- Java's System.out.print(a) + println(s)
    print!("{i}");
    println!("{s}");

    // Same line with explicit space
    println!("{i} {s}");

    Ok(())
}

/*
 * NOTES:
 *  - Rust enforces ownership rules: `s` is owned by this function — we converted
 *    a borrowed &str into String via to_string().
 *  - You CANNOT mix types in println! without a format placeholder for each.
 *  - For high-performance batch input, prefer `read_to_string` + token iteration.
 */
